src/Symfony/Bundle/TwigBundle/DependencyInjection/Compiler/ExceptionListenerPass.php

```diff
@@ -27,13 +29,18 @@ public function process(ContainerBuilder $container)
             return;
         }
 
-        // register the exception controller only if Twig is enabled and required dependencies do exist
-        if (!class_exists('Symfony\Component\ErrorRenderer\Exception\FlattenException') || !interface_exists('Symfony\Component\EventDispatcher\EventSubscriberInterface')) {
+        // to be removed in 5.0
+        // register the exception listener only if it's currently used, else use the provided by FrameworkBundle
+        if ('twig.controller.exception::showAction' === $container->getParameter('twig.exception_listener.controller') && $container->hasDefinition('exception_listener')) {
```

Comment:
`twig.controller.exception::showAction` is the default exception controller. How does this maintain BC if it changes the exception controller in the default config?

Reply:
If peoples aren't using a custom exception controller, why do we need to maintain the legacy behavior? Maybe I'm missing something... :thinking:

Reply:
Mmm custom non-HTML errors twig-based templates, yap, those are only supported by the legacy behavior. Searching another solution, thanks!

Reply:
what might work is removing the default value in the new framwork bundle config. if it is set, then ignore the twig bundle config. people upgrading need to explicitly add the new value on the framework bundle to opt-in to the new behavior. we can add this as a default to the reciepes for new projects.

Reply:
Fixed, now people will need set it to `null` explicitly to disable it and enable the framework error controller feature.

Reply:
then with https://github.com/symfony/recipes/pull/640 we're done for new 4.4 projects